README.md

```diff
@@ -9,3 +9,72 @@ primality proofs, commonly generated by the [Primo](http://www.ellipsa.eu/public
 application.
 
 Read more on [primality testing](http://cr.yp.to/primetests.html)
+
+# Installation
+
+Clone this repository:
+
+    git clone https://github.com/tomato42/ecpp-verifier.git
+    cd ecpp-verifier
+
+Install dependencies (for example on Fedora):
+
+    dnf install python3-ecdsa
+
+Run `ecpp` for the first time:
+
+    ./ecpp --help
+
+# Usage
+
+## Verifying primes in OpenSSH moduli file
+
+To verify all the primes in your OpenSSH moduli file have primality certificate,
+you can use ecpp this way:
+
+    ./ecpp -m /etc/ssh/moduli
+
+This will succeed for example for moduli file released with OpenSSH 8.2p1,
+listing certificates for each prime.
+
+## Generating primality certificates for primes in moduli file for OpenSSH
+
+If there are some primes without primality certificates, you can generate
+input files for Primo into `in/` directory.
+
+    ./ecpp -m /etc/ssh/moduli -p
+
+Now, open Primo downloaded from link above, extract archive and start GUI on
+a reasonably powerful machine (at this moment, Primo can work with up to
+64 cores).
+
+ * From *Menu*, select *Configure*, set number of cores your system have
+   (multithreading is not much useful)
+ * In tab *Certification*, select *3000 dd* (decimal digits) in *Trial
+   Division Parameters* and click *Build prime table* button.
+ * THen click *Load*. Select all the `.in` files in
+   the `in/` directory created by previous step and click *Open*.
+ * After long time, you will get certificates in `*.out` files in the same
+   directory.
+
+## Verify primality certificates
+
+In previous step, we got certificates for primes. Now we need to verify them.
+This can be done with the following command for one certificate:
+
+    ./ecpp -i in/primo-B412D0397A9D9-07E.out
+
+The job can be simply parallelized so if we want to verify all
+the certificates we got, we can use GNU parallel to get results in parallel,
+in this example using 16 parallel processes:
+
+    parallel -j16 "echo {} && ./ecpp -i {}" ::: in/*.out
+
+Now, we can add the primality certificates to `certificates/` directory
+
+## Verifying primes in OpenSSH moduli file again
+
+Running `ecpp` again as in the first example should give us certificate for
+each prime in the moduli file, confirming these are really primes.
```

Comment:
technically, this confirms that we have certificates for all primes in the moduli file

Reply:
Thanks. Fixed in current version.